shared-module/displayio/display_core.c

```diff
@@ -162,7 +162,7 @@ void displayio_display_core_set_rotation(displayio_display_core_t *self,
     }
 }
 
-bool displayio_display_core_show(displayio_display_core_t *self, displayio_group_t *root_group) {
+bool displayio_display_core_set_root_group(displayio_display_core_t *self, displayio_group_t *root_group) {
 
     if (root_group == NULL) { // set the display to the REPL, reset REPL position and size
```

Comment:
What do you think about making root_group == NULL not show anything? `show()` could keep this behavior but `root_group` would distinguish between `None` and `CIRCUITPYTHON_TERMINAL`.

Reply:
I do like that idea. From person using the API perspective that feels like the expected behavior. 

The first way that comes to mind for me to show nothing is to make a new empty Group and show that. This would visually appear like showing nothing, but is perhaps against the spirit of "Nothing" since it's also just a Group that you can then start adding things to. 

Is there some better or more proper way? If it is possible would we want to actually remove the old group and just not add a new one? Or is the empty Group sufficient?

Reply:
I think you'd actually want `None`. You can always add an empty Group from user code if that's what you want. (This might be trickier internally and you could hide the empty internal Group its easiest. `.root_group` should be `None` though.)

I also wonder if `None` shouldn't trigger the terminal when user code is done. That'd make disabling the terminal easier (https://github.com/adafruit/circuitpython/issues/2791).

Reply:
We may need to also make a check specifically for setting to `CIRCUITPYTHON_TERMINAL` as well to do this setup that sets the positon and starts the terminal. 

https://github.com/FoamyGuy/circuitpython/blob/ef3398422a88f72cf5863e768f6a5f705d1ffb9e/shared-module/displayio/display_core.c#L168-L173

I will give try it out with `None` and see if I can get that working. 

Do you know where to look to find the code causes the initialization of the terminal for the first time when the device boots up? I think something somewhere may call show(None) to get the terminal showing after boot up, if so it'll need to set it to CIRCUITPYTHON_TERMINAL instead.

Reply:
> Do you know where to look to find the code causes the initialization of the terminal for the first time when the device boots up? I think something somewhere may call show(None) to get the terminal showing after boot up, if so it'll need to set it to CIRCUITPYTHON_TERMINAL instead.

I think the first time is in `common_hal_displayio_display_construct` (`board_init` calls it.) Subsequent sets back to terminal are done in `reset_displays` I believe.